src/blockdata/script.rs

```diff
@@ -494,6 +572,27 @@ impl Script {
             None
         }
     }
+
+    /// Return type string.
+    pub fn type_string(&self) -> String {
+        if self.is_p2pk() {
+            "pubkey".to_string()
+        } else if self.is_p2pkh() {
+            "pubkeyhash".to_string()
+        } else if self.is_multisig() {
+            "multisig".to_string()
+        } else if self.is_p2sh() {
+            "scripthash".to_string()
+        } else if self.is_op_return() {
+            "nulldata".to_string()
+        } else if self.is_cp2pkh() {
+            "coloredpubkeyhash".to_string()
+        } else if self.is_cp2sh() {
+            "coloredscripthash".to_string()
+        } else {
+            "nonstandard".to_string()
+        }
+    }
```

Comment:
`enum AddressType` is defined in `src/util/address.rs` and `Display` trait for string notation is implemented. How about extends it?

Reply:
This method(type_string) is also intended to return script types including non-p2pkh or non-p2sh such as 'OP_RETURN ...'., while AddressType can not be defined for non-standard or op_return script.

Tapyrus-Core implementation: https://github.com/chaintope/tapyrus-core/blob/0a69fa72d394d7cf9119a718f8385f88c5323866/src/script/standard.cpp#L32

Reply:
(But we should update AddressType later because AddressType doesnt include colored coin address)

Reply:
> for non-standard or op_return script.

Right. Is it a script type rather than an address type?

Reply:
Add struct `ScriptType` (cf188c4)